### PR TITLE
Replace of seg_ms_lesion_mp2rage by UNIseg model on sct_deepseg

### PR DESCRIPTION
**Pull request by Nilser3**
# Description 

This PR replace the current `seg_ms_lesion_mp2rage` model by `UNIseg` (single-class 3D nnUnet model discused here #75 ) keeping the same name:  `seg_ms_lesion_mp2rage` (after comparisons these models see #81 )
Updates also the README file, explaining how to use the model, datasets for training and acknowledgments to the collaborators.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,119 +1,46 @@
...
-```
-
-> **Warning**
-> When running the MS lesion segmentation model, the image first need to be cropped around the spinal cord with a dilation of 32 in the axial plane.
-
-## Ensemble
+## Acknowledgments
 
-Multiple models were trained with different seeds, and they can all be used at test-time to produce slightly different segmentations that can then be everaged (ensemble/bagging). See more details [here](https://github.com/ivadomed/model_seg_ms_mp2rage/issues/63#issuecomment-1409250538).
+Charidimos Tsagkas and Daniel Reich from Translational Neuroradiology Section, National Institutes of Health (NIH), Bethesda, USA, Cristina Granziera from Neurologic Clinic and Policlinic, University Hospital Basel and University of Basel, Basel, Switzerland and Virginie Callot from Center for Magnetic Resonance in Biology and Medicine (CRMBM-CEMEREM, UMR 7339, CNRS, Aix-Marseille University, Marseille, France).
```
Comment (jcohenadad) [inline README.md]:
```suggestion
- Charidimos Tsagkas (Translational Neuroradiology Section, National Institutes of Health, Bethesda, USA)
- Daniel Reich (Translational Neuroradiology Section, National Institutes of Health, Bethesda, USA)
- Cristina Granziera (Neurologic Clinic and Policlinic, University Hospital Basel and University of Basel, Basel, Switzerland)
- Virginie Callot (Center for Magnetic Resonance in Biology and Medicine, CRMBM-CEMEREM, UMR 7339, CNRS, Aix-Marseille University, Marseille, France)
```

File: README.md [hunk 2]
```diff
@@ -1,119 +1,46 @@
...
 ## Model overview 
 
-![model_overview](https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20211223/model_overview.png)
+![uniseg](https://github.com/ivadomed/model_seg_ms_mp2rage/assets/77469192/d7fd985b-5c32-43bb-931f-e9c114a98b4c)
 
-As of https://github.com/ivadomed/model_seg_ms_mp2rage/releases/tag/r20230210, five models are trained (with dropout and different seeds), to perform ensemble/bagging during inference and compute epistemic uncertainty.
+3D model trained with [nnUNetv2](https://github.com/MIC-DKFZ/nnUNet) framework.
 
 ## Dependencies
+[SCT 6.3](https://spinalcordtoolbox.com/)
```
Comment (jcohenadad) [inline README.md]:
```suggestion
[SCT 6.4](https://spinalcordtoolbox.com/)
```

File: README.md [hunk 3]
```diff
@@ -1,119 +1,46 @@
...
+**Warning:** When running the MS lesion segmentation model, the image first need to be cropped around the spinal cord mask with a dilation of 30 mm in the axial plane and 5 mm in the Z-axis. 
 
-Spinal cord segmentation training was carried out with
+## Launch MS lesion segmentation:
+1. Spinal cord segmentation 
+```bash
+sct_deepseg -i IMAGE_UNIT1 -task seg_sc_contrast_agnostic -o IMAGE_UNIT1_sc
 ```
-ivadomed --train -c config/seg_sc.json
+2. Cropping with dilation (for images with 1mm isotropic resolution)
```
Comment (jcohenadad) [inline README.md]:
```suggestion
2. Cropping with dilation (for axial orientation images with 1mm isotropic resolution)
```

**Review by jcohenadad — CHANGES_REQUESTED** [review]
@Nilser3 indicate where to find model to test

**Comment by Nilser3** [comment]
> @Nilser3 indicate where to find model to test

Thanks you @jcohenadad 
The model is here https://github.com/ivadomed/model_seg_ms_mp2rage/releases/tag/r20240610 

I have already replaced the `seg_ms_lesion_mp2rage` model and it is ready to be tested this branch:

branch on SCT: `nlm/update-_seg_ms_lesion_mp2rage-model`
commit: `b64340b4259aa99734a4ef809edf4af882261409`

**Comment by jcohenadad** [comment]
Getting this message during inference:
```console
/Users/julien/code/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/nnunetv2/utilities/plans_handling/plans_handler.py:37: UserWarning: Detected old nnU-Net plans format. Attempting to reconstruct network architecture parameters. If this fails, rerun nnUNetv2_plan_experiment for your dataset. If you use a custom architecture, please downgrade nnU-Net to the version you implemented this or update your implementation + plans.
  warnings.warn("Detected old nnU-Net plans format. Attempting to reconstruct network architecture "
```
any concerns?

**Comment by jcohenadad** [comment]
Another message, possibly concerning:
```console
Found Intel OpenMP ('libiomp') and LLVM OpenMP ('libomp') loaded at
the same time. Both libraries are known to be incompatible and this
can cause random crashes or deadlocks on Linux when loaded in the
same Python program.
Using threadpoolctl may cause crashes or deadlocks. For more
information and possible workarounds, please see
    https://github.com/joblib/threadpoolctl/blob/master/multiple_openmp.md

```

**Review by jcohenadad — CHANGES_REQUESTED** [review]
The [zipped model](https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20240610/model_ms_lesion_mp2rage-unit1_3d.zip) does not seem to include the list of subjects used for training and the version of the dataset used to train the model. It is required.

**Comment by jcohenadad** [comment]
Inference time is pretty long... I've tried without cropping on the basel data, inference time was 4min 6s.

With cropping: 7s 😊

**Comment by jcohenadad** [comment]
Related question: the progress bar shows 100 iterations:
```console
 19%|█████████████████████████████▊                                                                                                                               | 19/100 [00:42<02:52,  2.14s/it]
```
What to these correspond to? And how come the number of iterations is the same across different input volume size?

**Comment by jcohenadad** [comment]
Comparison with previous model, and current model with various crop sizes:

ezgif-6-1f1309b0b4

Conclusion: too narrow cropping (5x5x5) leads to bad segmentation. A minimum cropping size is required. What size? @Nilser3 that's a follow-up investigation

**Comment by jcohenadad** [comment]
> Conclusion: too narrow cropping (5x5x5) leads to bad segmentation. A minimum cropping size is required. What size? @Nilser3 that's a follow-up investigation

follow-up idea: given that the inference is _sliiiiightly_ different across crop sizes, how about computing a soft segmentation based on the inference performed across various crop sizes?

**Comment by Nilser3** [comment]
> The [zipped model](https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20240610/model_ms_lesion_mp2rage-unit1_3d.zip) does not seem to include the list of subjects used for training and the version of the dataset used to train the model. It is required.

Thank you @jcohenadad,
Ok, the [release r20240610](https://github.com/ivadomed/model_seg_ms_mp2rage/releases/tag/r20240610) has been updated with the needed information!

**Comment by jcohenadad** [comment]
@Nilser3 any update? this is becoming urgent: https://forum.spinalcordmri.org/t/segmentation-of-spinal-cord-lesion-on-mp2rage-uni-images-doesnt-work/1302

**Comment by Nilser3** [comment]
> @Nilser3 any update? this is becoming urgent: https://forum.spinalcordmri.org/t/segmentation-of-spinal-cord-lesion-on-mp2rage-uni-images-doesnt-work/1302

The model was replaced in the [release 6.4](https://github.com/spinalcordtoolbox/spinalcordtoolbox/releases/tag/6.4), this branch is ok for PR.

**Pull request merged**